src/Stratis.Bitcoin.Features.SmartContracts/PoA/WhitelistedHashChecker.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections.Generic;
+using NBitcoin;
+using Stratis.Bitcoin.Features.PoA.Voting;
+
+namespace Stratis.Bitcoin.Features.SmartContracts.PoA
+{
+    /// <summary>
+    /// Checks hashes against a whitelist.
+    /// </summary>
+    public class WhitelistedHashChecker : IWhitelistedHashChecker
+    {
+        private readonly IWhitelistedHashesRepository whitelistedHashesRepository;
+
+        public WhitelistedHashChecker(IWhitelistedHashesRepository whitelistedHashesRepository)
+        {
+            this.whitelistedHashesRepository = whitelistedHashesRepository;
+        }
+
+        /// <summary>
+        /// Checks that a supplied hash is present in the whitelisted hashes repository.
+        /// </summary>
+        /// <param name="hash">The bytes of the hash to check.</param>
+        /// <returns>True if the hash was found in the whitelisted hashes repository.</returns>
+        public bool CheckHashWhitelisted(byte[] hash)
+        {
+            if (hash.Length != 32)
+            {
+                // For this implementation, only 32 byte wide hashes are accepted.
+                return false;
```

Comment:
Please add exit log (but up to you, I don't know if FN policies apply atm or not)